### PR TITLE
Fixes issues with neuron and corenrn in neurodamus

### PR DESCRIPTION
**Pull request by iomaganaris**
- Fixes issue with installing neurodamus with neuron+binary
- Fixes issue with running nrnivmodl-core outside spack recipe

### CODE REVIEW

**Comment by pramodk** [comment]
LGTM 
I would wait for @ferdonline  to review sim-model change next week.
@ferdonline  : we are still using `neuron~binary` but the idea to remove that constraint is that if I want to build with `neuron+binary` during debugging, I should be able to do it.

**Comment by iomaganaris** [comment]
Hey @ferdonline. Can you take a look when you have time? :innocent:

File: var/spack/repos/builtin/packages/coreneuron/package.py [hunk 1]
```diff
@@ -150,6 +150,21 @@ def cmake_args(self):
...
         return options
 
+    @run_after('install')
+    def filter_compilers(self):
+        """run after install to avoid spack compiler wrappers
+        getting embded into nrnivmodl script"""
+
+        nrnmakefile = join_path(self.prefix, 'share/coreneuron/nrnivmodl_core_makefile')
+
+        kwargs = {
```
Comment (ferdonline) [inline var/spack/repos/builtin/packages/coreneuron/package.py:160]: Apart from using many lines, this is nice :)

**Review by ferdonline — APPROVED** [review]
I was concerned that eventually were had more dependencies on the shared libs, but it seems not and also the two created here are independent, so all seems good

**Pull request merged**